File: leetcode/python/0495_teemo_attacking.py
class Solution:
    def findPoisonedDuration(self, timeSeries: List[int], duration: int) -> int:
        # res, endTime = -1
        # loop through timeSeries using i:
        # if timeSeries[i] <= endTime: res -= (endTime - timeSeries[i]) + 1, res += duration, endTime = timeSeries[i] + duration - 1
        # elif timeSeries[i] > endTime: res += duration, endTime = timeSeries[i] + duration - 1 
        res, endTime = 0, -1
        for i in range(0, len(timeSeries)):
            if timeSeries[i] <= endTime:
                res -= (endTime - timeSeries[i]) + 1
                res += duration
            elif timeSeries[i] > endTime:
                res += duration

            endTime = timeSeries[i] + duration - 1

        return res

File: leetcode/java/0495_teemo_attacking.java
class Solution {
    public int findPoisonedDuration(int[] timeSeries, int duration) {
        int res = 0, prevEnd = Integer.MIN_VALUE;

        for (int i = 0; i < timeSeries.length; i++) {
            if (prevEnd >= timeSeries[i]) {
                res += duration - (prevEnd - timeSeries[i] + 1);
            } else {
                res += duration;
            }
            prevEnd = timeSeries[i] + duration - 1;
        }

        return res;
    }
}